ui/src/app/utils/index.ts

```diff
@@ -12,3 +14,15 @@ export function isBlank(toTest: String): boolean {
 export function deepCopy(obj: Object): Object {
   return fromJS(obj).toJS();
 }
+
+export function flattenedRouteData(route: ActivatedRoute): Observable<Data> {
```

Comment:
Ooh I don't know if I like this.  What this does is flatten the snapshots of a given route, but it returns an observable like you've flattened the _observables_ that the snapshots are temporary views into.  Imho this should either return just the data (not wrapped in an observable) or should flatten the actual observables like this:

```
const datas = route.pathFromRoot.map(route => route.data);
const merger = (objs) => objs.reduce((acc, obj) => ({...acc, ...obj}), {});
return combineLatest(datas, merger);
```

Reply:
Thanks John - I was intending to ask for your opinion on this. It will need to be accessed by different components (just pushed something for the demographics component as well). I kept it as an observable to minimize the changes needed to cohort-search. I can certainly take it out of that structure but that changes the cohort-search logic a little bit. 

Also - I'm still not finished with looking for all of the places where we are assuming workspace is on the snapshot data. I think it's in a couple other places I haven't looked at yet.